docs/geometry.rst

```diff
@@ -49,11 +58,23 @@ The three shapes supported are :code:`Circle`, :code:`Ellipse`, and :code:`Polyg
                               value=8)
 
 
+**Box**
+
+.. code-block :: python
+
+    # All units are decimal degrees
+    # A box differs from a polygon in that the sides will be constant ra/dec rather
+    # than great circles.
+    # See https://hpgeom.readthedocs.io/en/latest .
+    box = healsparse.Box(ra1=20.0, ra2=30.0, dec1=10.0, dec2=5.0, value=True)
+
+
 Combining Geometric Objects with Maps
 -------------------------------------
 
 Given a map, it is very simple to combine geometric objects to build up complex shapes/masks/etc.
 Behind the scenes, large geometric objects are rendered with :code:`hpgeom` pixel ranges which leads to greater memory efficiency.
+Note that these operations can be applied to integer or boolean maps.
```

Comment:
What happens if one mixes a boolean `value` for the geometry but applies it to an integer map or vice versa? Does healsparse do casts so that 0 is equivalent False and non-zero is True and True casts to 1?

This part of the API confuses me.

Reply:
Oh this is an interesting one.  The design of healsparse is that it doesn't want to do any implicit conversion.  Certainly it doesn't allow it between numpy types -- you have to explicitly cast.  The problem is that if you use a python type it does allow implicit conversions.  I might need to fix that (would like to hear your thoughts).  For example, if you have an integer map `np.int64` and you try to add in a circle with `value=True` then it'll promote that to `1` (default numpy/python promotion).  But if you do `value=np.bool_(True)` then it raises because of the type mismatch (which is checked when you have two numpy dtypes).

Reply:
OK. What about bit packed boolean arrays? That case is ambiguous to me since I cannot specify an explicitly bit-packed boolean value for the geometry.

Reply:
They work the same way.  I know you can't specify a bit packed map for `get_map` but you really should never need `get_map` for anything.  You generate a bit packed or whatever map and then just `|=` or `&=` your shapes into it with a `True` or `False` value.

Reply:
This snippet might help:

```
np.array(True).dtype == np.bool_
```

It could be easiest to just convert to numpy right at the start?

Reply:
```
In [2]: import numpy as np

In [3]: np.array(True).dtype == np.bool_
Out[3]: True

In [4]: np.array(0).dtype
Out[4]: dtype('int64')

In [5]: np.array(1).dtype
Out[5]: dtype('int64')
```

Reply:
#arrayscalarsforthewin?

Reply:
One of the things that I want to preserve is allowing the use of (e.g.) a python int to set an int32 map.  But I need to add range checking.  So this will be more work than just in geom to make sure this works right.

Reply:
Sounds good. Let's punt on this issue then. I made on suggestion below to keep the code function for specifying `dtype=bool` as opposed to `dtype=np.bool_` in `get_map`. With that change, we should be good to go.